src/inject/dynamic-theme/is-it-dark.ts

```diff
@@ -0,0 +1,71 @@
+import type {RGBA} from '../../utils/color';
+import {rgbToHSL} from '../../utils/color';
+import {tryParseColor} from './modify-css';
+import type {VariablesStore} from './variables';
+
+const IS_DARK_TRESHOLD = 0.5;
+
+let bodyObserver: MutationObserver;
+const bodyCallbacks = new Set<() => void>();
+
+async function waitTillBody(): Promise<void> {
+    return new Promise<void>((resolve) => {
+        if (!bodyObserver) {
+            resolve();
+        }
+        bodyCallbacks.add(resolve);
+    });
+}
+
+export function IsItDark(variableStore: VariablesStore, selectorText: string, styleRule: CSSStyleDeclaration): boolean | Promise<boolean> {
```

Comment:
Probably better to have three classes of output, dark, medium, and light.

Reply:
I'm not sure. What should DR do on the "medium" output?

Reply:
Do nothing, because the page does not make enough sense.

Reply:
That is the behavior of `dark` output. So Ill prefer to stay with a simple boolean as output.